pkg/sfu/buffer/buffer.go

```diff
@@ -328,41 +344,58 @@ func (b *Buffer) calc(pkt []byte, arrivalTime int64) {
 func (b *Buffer) updateStreamState(p *rtp.Packet, pktSize int, arrivalTime int64, isRTX bool) {
 	sn := p.SequenceNumber
 
-	if b.stats.PacketCount == 0 {
-		b.highestSN = sn - 1
+	if !b.started {
+		b.started = true
+		b.highestSN = sn
 
 		b.lastReport = arrivalTime
 
 		b.rrSnapshot = &receiverReportSnapshot{
-			extSeqNum:       uint32(sn) - 1,
-			packetsReceived: 0,
-			packetsLost:     0,
-			lastLossRate:    0.0,
-		}
-	}
-
-	diff := sn - b.highestSN
-	if diff > (1 << 15) {
-		// out-of-order, remove it from nack queue
-		if b.nacker != nil {
-			b.nacker.Remove(sn)
+			extHighestSeqNum: uint32(sn) - 1,
+			packetsLost:      0,
+			lastLossRate:     0.0,
 		}
 	} else {
-		if b.nacker != nil && diff > 1 {
-			for lost := b.highestSN + 1; lost != sn; lost++ {
-				b.nacker.Push(lost)
+		diff := sn - b.highestSN
+		if diff > (1 << 15) {
+			if !isRTX && b.stats.TotalPacketsLost != 0 {
+				b.stats.TotalPacketsLost--
 			}
-		}
 
-		if sn < b.highestSN && b.stats.PacketCount > 0 {
-			b.cycle++
-		}
+			// out-of-order, remove it from nack queue
+			if b.nacker != nil {
+				b.nacker.Remove(sn)
+			}
+		} else {
+			b.stats.TotalPacketsLost += (uint32(diff) - 1)
+			if b.nacker != nil && diff > 1 {
+				for lost := b.highestSN + 1; lost != sn; lost++ {
+					b.nacker.Push(lost)
+				}
+			}
 
-		b.highestSN = sn
+			if sn < b.highestSN {
+				b.cycle++
+			}
+
+			b.highestSN = sn
+		}
 	}
 
-	b.stats.PacketCount++
-	b.stats.TotalBytes += uint64(pktSize)
+	switch {
+	case isRTX:
+		b.stats.TotalRetransmitPackets++
+		b.stats.TotalRetransmitBytes += uint64(pktSize)
+	case len(p.Payload) == 0:
+		b.stats.TotalPaddingPackets++
+		b.stats.TotalPaddingBytes += uint64(pktSize)
+	default:
+		b.stats.TotalPrimaryPackets++
+		b.stats.TotalPrimaryBytes += uint64(pktSize)
+		if p.Marker {
+			b.stats.TotalFrames++
+		}
```

Comment:
Measuring total number of frames. For some reason, for audio tracks this number is very small from Chrome. Need to dig in. Video counts seem to be fine. But, need to see more data (maybe graphed so it is easier to see)

Reply:
I think that's fine, we don't need frames for audio, fps is used only for video tracks.